java/arcs/core/data/MuxType.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+import arcs.core.type.Tag
+import arcs.core.type.Type
+import arcs.core.type.TypeFactory
+import arcs.core.type.TypeLiteral
+
+data class MuxType<T : Type>(private val innerType: T) :
```

Comment:
KDoc would be very valuable here. Explaining what a MuxType is, where/how it is used.